File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans=0;
        unordered_set<int>s(nums.begin(),nums.end());
        for(int num:s){
            if(s.count(num-1)==0){
                int curr=num;
                int count=1;
                while(s.count(curr+1)){
                    curr++;
                    count++;
                }
                ans=max(ans,count);
            }
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer>st=new HashSet<>();
        for(int num:nums)st.add(num);
        int maxLen=0;
        for(int num:nums){
            if(!st.contains(num-1)){
                int curr=num;
                int count=1;
                while(st.contains(curr+1)){
                    curr++;
                    count++;
                }
                maxLen=Math.max(maxLen,count);
            }
            
        }
        return maxLen;
    }
}